### PR TITLE
Pretty json

### PR DESCRIPTION
**Pull request by BVMiko**
#### Which issue(s) does this change fix?
#3577 

#### Why is this change necessary?
The default format is hard to manually parse

#### How does it address the issue?
It modifies the StreamWriter, adding an optional argument to filter the stream content.  It also modifies the API Gateway service implementation to make stack traces more readable.

#### What side effects does this change have?
The primary changes are to the output displayed to the screen (or optionally logged to a file).  Additionally, there is a minor change in the determination of whether or not a Lambda response is an error.

#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [x] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by hawflau** [comment]
Hi @BVMiko Thanks for the PR! I'll review the changes. Meanwhile, can you check the [checks](https://ci.appveyor.com/project/AWSSAMCLI/aws-sam-cli/builds/42179849) and fix the unit tests?

File: samcli/commands/local/cli_common/invoke_context.py [hunk 1]
```diff
@@ -387,7 +387,7 @@ def stderr(self) -> StreamWriter:
             Stream writer for stderr
         """
         stream = self._log_file_handle if self._log_file_handle else osutils.stderr()
-        return StreamWriter(stream, auto_flush=True)
+        return StreamWriter(stream, auto_flush=True, pretty_json=True)
```
Comment (hawflau) [inline samcli/commands/local/cli_common/invoke_context.py:390]: I think we should also offer an option for users to not prettify json, as expanding the JSON object does not always provide better UX.

**Review by hawflau — CHANGES_REQUESTED** [review]
Thanks again for the PR. 

It seems the pretty json only applies for Lambdas using JSON structured logging (e.g. with Lambda Powertools). Can you update your PR description to make this clear?

**Comment by jfuss** [comment]
Closing PR as this looks abandoned by the author.

@BVMiko If you still want to contribute this in the future and the issue hasn't been solved, you are welcome to open a new PR or this one. Thanks again for submitting and spending time on the PR.

**Pull request closed without merging**